src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentState.cs

```diff
@@ -160,12 +160,14 @@ public virtual DocumentState WithImportsChange()
         return state;
     }
 
-    public virtual DocumentState WithProjectChange()
+    public virtual DocumentState WithProjectChange(bool cacheComputedState)
```

Comment:
I'm still not sure when or why this should be true or false. I see in the PR but not really the impact of it

Reply:
I'll admit this isn't great, and I think the reasoning around what the computed state actually achieves, and whether its worth caching, is not fully understood by anyone. Or perhaps better said, I don't think I, or anyone else, could write a convincing comment explaining why this value should be true or false. I've made things match previous behaviour, and that seems to have had no bad outcomes, and I look forward to Dustin removing the computed state tracker in the future.